### PR TITLE
avm1: implemented mouse_wheel_enabled to text fields and made text scrollable

### PR DESCRIPTION
**Pull request by Flawake**
added mouse_wheel_enabled to text fields, made text scrollable and implemente onScroller for avm1.
closes: #14796

CC: #280

### CODE REVIEW

File: core/src/display_object/edit_text.rs [hunk 1]
```diff
@@ -1785,6 +1785,18 @@ impl<'gc> EditText<'gc> {
...
+    fn on_scroller(&self, activation: &mut Avm1Activation<'_, 'gc>) {
+        if let Avm1Value::Object(object) = self.object() {
+            let _ = object.call_method(
+                "broadcastMessage".into(),
+                &["onScroller".into(), object.into()],
+                activation,
+                ExecutionReason::Special,
+            );
+        }
+        //TODO: Implement this for Avm2
```
Comment (Lord-McSweeney) [inline core/src/display_object/edit_text.rs:1797]: Is this just the mouse wheel event?

Comment (michiel2005) [reply core/src/display_object/edit_text.rs:1797]: It is indeed for the mouse scroll event. Your PR does the AVM2 side of this, right?

Comment (adrian17) [reply core/src/display_object/edit_text.rs:1797]: > Is this just the mouse wheel event? 

AFAIK no, in AS3 a textfield has a separate `scroll` event.

That said, I don't think it's blocking this PR?
Though what probably should be investigated is the interaction of these two PRs, as in: whether capturing this event and `.preventDefault()`ing it is supposed to disable scrolling textfields.

Comment (Lord-McSweeney) [reply core/src/display_object/edit_text.rs:1797]: `MouseEvent.WHEEL` and `Event.SCROLL` cannot be `preventDefault`ed, so this should be fine.

**Pull request merged**